lib/kafo/base_context.rb

```diff
@@ -24,12 +26,18 @@ def self.clear_caches
 
     def self.facts
       @facts ||= begin
-        symbolize(JSON.load(`#{facter_path} --json`) || {})
+        result = run_command("#{facter_path} --json")
+        symbolize(JSON.load(result) || {})
       end
     end
 
     def self.facter_path
       @facter_path ||= PuppetCommand.search_puppet_path('facter')
     end
+
+    def self.run_command(command)
+      stdout, stderr, _status = Open3.capture3(command)
```

Comment:
Looking ahead, Facter 4 will be written in Ruby. Should we be future proof and clean the env already?

Reply:
I just forgot to put this back due to the testing I had to do to debug some issues.